e2-studio-star-rx72n-firmware/libs/rx_spi_link/src/rx_spi_link.c

```
@@ -0,0 +1,731 @@
+     *
+     * IMPORTANT: This formula is correct because rx_fec.c internally accounts
+     * for k_fec_tail_bits when encoding. The frame payload contains the full
+     * FEC-encoded bitstream (data bits + parity bits + tail bits). The formula:
+     *   expected_decoded_len = (soft_len / k_bits_per_byte) / 2
+     * works as follows:
+     *   1. soft_len is the total encoded bit count (includes tail bits)
+     *   2. Divide by k_bits_per_byte (8) to get encoded byte count
+     *   3. Divide by 2 because FEC rate is 1/2 (each input bit → 2 output bits)
+     * The FEC decoder (rx_fec_viterbi_decode) automatically handles tail bit
+     * flushing internally, so we don't subtract them here. */
+  const uint32_t expected_decoded_len = (soft_len / (uint32_t)k_bits_per_byte) / 2U;
```

Clang (14.0.6)

[warning] variable 'expected_decoded_len' is not initialized

(cppcoreguidelines-init-variables)